projects/OG-Analytics/src/main/java/com/opengamma/analytics/financial/interestrate/future/provider/BondFutureOptionMarginSecurityBlackSmileMethod.java

```diff
@@ -0,0 +1,299 @@
+    final double[] priceAdjoint = BLACK_FUNCTION.getPriceAdjoint(option, dataBlack);
+    // Backward sweep
+    final double priceBar = 1.0;
+    final double priceFutureBar = priceAdjoint[1] * priceBar;
+    final MulticurveSensitivity priceFutureDerivative = METHOD_FUTURE.priceCurveSensitivity(
+        security.getUnderlyingFuture(), black.getIssuerProvider());
+    return priceFutureDerivative.multipliedBy(priceFutureBar);
+  }
+
+  /**
+   * Computes the option security price curve sensitivity. 
+   * It is supposed that for a given strike the volatility does not change with the curves.
+   * The option price and its derivative wrt the futures price is computed using the futures price. 
+   * The derivatives of the futures price with respect to the curves are computed using the curves.
+   * @param security The future option security, not null
+   * @param black The curve and Black volatility data, not null
+   * @return The security price curve sensitivity.
+   */
+  public MulticurveSensitivity priceCurveSensitivity(BondFuturesOptionMarginSecurity security,
+      BlackBondFuturesProviderInterface black) {
+    ArgumentChecker.notNull(security, "security");
+    ArgumentChecker.notNull(black, "Black data");
+    double priceFutures = METHOD_FUTURE.price(security.getUnderlyingFuture(), black.getIssuerProvider());
+    return priceCurveSensitivityFromPrice(security, black, priceFutures);
+  }
+
+  /**
+   * Interpolates and returns the option's implied volatility 
+   * @param security The future option security, not null
+   * @param black The curve and Black volatility data, not null
+   * @return Lognormal Implied Volatility.
+   */
+  public double impliedVolatility(final BondFuturesOptionMarginSecurity security,
+      final BlackBondFuturesProviderInterface black) {
+    ArgumentChecker.notNull(security, "security");
+    ArgumentChecker.notNull(black, "Black data");
+    final double priceFutures = METHOD_FUTURE.price(security.getUnderlyingFuture(), black.getIssuerProvider());
+    final double strike = security.getStrike();
+    final double delay = security.getUnderlyingFuture().getNoticeLastTime() - security.getExpirationTime();
+    final double volatility = black.getVolatility(security.getExpirationTime(), delay, strike, priceFutures);
+    return volatility;
+  }
+
+  /**
+   * Computes the underlying future security price. 
+   * @param security The future option security, not null
+   * @param issuerMulticurves Issuer and multi-curves provider.
+   * @return The security price.
+   */
+  public double underlyingFuturePrice(final BondFuturesOptionMarginSecurity security,
+      final IssuerProviderInterface issuerMulticurves) {
+    ArgumentChecker.notNull(security, "security");
+    return METHOD_FUTURE.price(security.getUnderlyingFuture(), issuerMulticurves);
+  }
+
+  /**
+   * The delta of the option without the underlying futures price.
```

Comment:
Change the description: "The theoretical delta in the Black model. The underlying futures price is computed from the curves." Or something similar. Same for gamma, vega and theta.